			     BASH PATCH REPORT
			     =================

Bash-Release:	5.2
Patch-ID:	bash52-009

Bug-Reported-by:	Corey Hickey <bugfood-ml@fatooh.org>
Bug-Reference-ID:	<134330ef-0ead-d73e-68eb-d58fc51efdba@fatooh.org>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/help-bash/2022-10/msg00025.html

Bug-Description:

Bash arithmetic expansion should allow `@' and `*' to be used as associative
array keys in expressions.

Patch (apply with `patch -p0'):

*** ../bash-5.2-patched/expr.c	2022-07-11 10:03:34.000000000 -0400
--- expr.c	2022-10-31 10:51:08.000000000 -0400
***************
*** 1169,1172 ****
--- 1169,1174 ----
  #if defined (ARRAY_VARS)
    aflag = tflag;	/* use a different variable for now */
+   if (shell_compatibility_level > 51)
+     aflag |= AV_ATSTARKEYS;
    v = (e == ']') ? array_variable_part (tok, tflag, (char **)0, (int *)0) : find_variable (tok);
  #else
*** ../bash-5.2/patchlevel.h	2020-06-22 14:51:03.000000000 -0400
--- patchlevel.h	2020-10-01 11:01:28.000000000 -0400
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 8
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 9
  
  #endif /* _PATCHLEVEL_H_ */
